tests/mgbench/benchmark.py

```diff
@@ -29,6 +31,151 @@
 import log
 import runners
 
+WITH_FINE_GRAINED_AUTHORIZATION = "with_fine_grained_authorization"
+WITHOUT_FINE_GRAINED_AUTHORIZATION = "without_fine_grained_authorization"
+
+# Parse options.
+parser = argparse.ArgumentParser(
+    description="Memgraph benchmark executor.", formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+parser.add_argument(
+    "benchmarks",
+    nargs="*",
+    default="",
+    help="descriptions of benchmarks that should be run; "
+    "multiple descriptions can be specified to run multiple "
+    "benchmarks; the description is specified as "
+    "dataset/variant/group/query; Unix shell-style wildcards "
+    "can be used in the descriptions; variant, group and query "
+    "are optional and they can be left out; the default "
+    "variant is '' which selects the default dataset variant; "
+    "the default group is '*' which selects all groups; the"
+    "default query is '*' which selects all queries",
+)
+parser.add_argument(
+    "--vendor-binary",
+    help="Vendor binary used for benchmarking, by defuault it is memgraph",
+    default=helpers.get_binary_path("memgraph"),
+)
+
+parser.add_argument(
+    "--vendor-name",
+    default="memgraph",
+    help="Input vendor binary name (memgraph, neo4j)",
+)
+parser.add_argument(
+    "--client-binary",
+    default=helpers.get_binary_path("tests/mgbench/client"),
+    help="Client binary used for benchmarking",
+)
+parser.add_argument(
+    "--num-workers-for-import",
+    type=int,
+    default=multiprocessing.cpu_count() // 2,
+    help="number of workers used to import the dataset",
+)
+parser.add_argument(
+    "--num-workers-for-benchmark", type=int, default=1, help="number of workers used to execute the benchmark"
+)
+
+parser.add_argument(
```

Comment:
Explain the motivation to have this flag and general tradeoff between counting queries and timing the execution